### PR TITLE
🧹 core-plugin-api: Deprecate `favoriteEntityTooltip` & `favoriteEntityIcon`

### PR DESCRIPTION
**Pull request by jhaals**
Deprecated `favoriteEntityTooltip` and `favoriteEntityIcon` since the utility value is very low.

### CODE REVIEW

File: plugins/catalog/src/components/CatalogTable/CatalogTable.tsx [hunk 1]
```diff
@@ -116,8 +123,8 @@ export const CatalogTable = (props: CatalogTableProps) => {
       const isStarred = isStarredEntity(entity);
       return {
         cellStyle: { paddingLeft: '1em' },
-        icon: () => favoriteEntityIcon(isStarred),
-        tooltip: favoriteEntityTooltip(isStarred),
+        icon: () => (isStarred ? <YellowStar /> : <StarBorder />),
```
Comment (freben) [inline plugins/catalog/src/components/CatalogTable/CatalogTable.tsx:126]: i'd suggest `icon: () => <FavoriteEntity starred={isStarred} />` and introducing that as an optional force-set-to-true-or-false prop. It even has the tooltip, so you can remove that from the next line

same below obvs

Comment (Rugvip) [reply plugins/catalog/src/components/CatalogTable/CatalogTable.tsx:126]: Btw, I think the proper way forward for these might be to add them as system icons, later x)

Comment (Rugvip) [reply plugins/catalog/src/components/CatalogTable/CatalogTable.tsx:126]: Yeah could do something like that

Comment (Rugvip) [reply plugins/catalog/src/components/CatalogTable/CatalogTable.tsx:126]: Let's merge as is and figure out the way forward in an issue

Comment (Rugvip) [reply plugins/catalog/src/components/CatalogTable/CatalogTable.tsx:126]: #9948

**Pull request merged**